File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        tail = head
        while head and head.next:
            if head.val != head.next.val:
                head = head.next
            else:
                head.next = head.next.next
        return tail
                
            
        

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* prev = head;
        if(head == nullptr){
            return head;
        }
        ListNode* curr = head->next;
        while(curr){
            if(prev->val == curr->val){
                prev->next = curr->next;
                curr = curr->next;
            }
            else{
                prev = curr;
                curr = curr->next;
                
            }
        }
        return head;
        }
        
};